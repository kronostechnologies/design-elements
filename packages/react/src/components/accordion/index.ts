export { Accordion, type ItemsProps } from './accordion';
export { accordionClasses, type AccordionClasses } from './accordion-classes';
