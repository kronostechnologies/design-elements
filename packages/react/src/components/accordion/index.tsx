export { Accordion } from './accordion';
export { AccordionContainer } from './accordion-container';
