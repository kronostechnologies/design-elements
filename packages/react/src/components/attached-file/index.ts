export { attachedFileClasses, AttachedFileClasses } from './attached-file-classes';
export { AttachedFile, AttachedFileProps } from './attached-file';
