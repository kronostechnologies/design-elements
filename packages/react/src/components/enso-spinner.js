"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var styled_components_1 = __importDefault(require("styled-components"));
var enso_svg_1 = __importDefault(require("../icons/enso.svg"));
var EnsoSpinner = styled_components_1.default(enso_svg_1.default)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  animation: roll 1s infinite;\n  animation-timing-function: linear;\n  fill: #e2732d;\n  height: 80px;\n  width: 83px;\n\n  @keyframes roll {\n    from { transform: rotate(0deg); }\n    to { transform: rotate(360deg); }\n  }\n"], ["\n  animation: roll 1s infinite;\n  animation-timing-function: linear;\n  fill: #e2732d;\n  height: 80px;\n  width: 83px;\n\n  @keyframes roll {\n    from { transform: rotate(0deg); }\n    to { transform: rotate(360deg); }\n  }\n"])));
exports.EnsoSpinner = EnsoSpinner;
var templateObject_1;
