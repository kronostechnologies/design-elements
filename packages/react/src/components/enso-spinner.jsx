import styled from 'styled-components';

import enso from '../icons/enso.svg';

export default styled(enso)`
  animation: roll 1s infinite;
  animation-timing-function: linear;
  fill: #e2732d;
  height: 80px;
  width: 83px;

  @keyframes roll {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  }
`;
