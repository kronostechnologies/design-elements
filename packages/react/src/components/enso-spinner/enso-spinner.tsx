import styled from 'styled-components';

import Enso from '../../icons/enso.svg';
import { equisoftTheme } from '../../themes/equisoft';
import { Theme } from '../theme-wrapper/theme-wrapper';

const EnsoSpinner = styled(Enso)`
  ${(props: {theme: Theme}) => {
      const theme = Object.entries(props.theme).length === 0 ? equisoftTheme : props.theme;
      return `
      animation: roll 1s infinite;
      animation-timing-function: linear;
      fill: ${theme.main['primary-1.1']};
      height: 80px;
      width: 83px;

      @keyframes roll {
        from { transform: rotate(0deg); }
        to { transform: rotate(360deg); }
      }
    `;
  }}
`;

export { EnsoSpinner };
