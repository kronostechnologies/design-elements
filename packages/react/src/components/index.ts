export * from './accordion';
export * from './avatar';
export * from './badge';
export * from './bento-menu-button';
export * from './breadcrumb';
export * from './buttons';
export * from './card';
export * from './card-link';
export * from './carousel';
export * from './checkbox';
export * from './checkbox-group';
export * from './chooser-button';
export * from './chooser-button-group';
export * from './combobox';
export * from './date-picker';
export * from './design-system';
export * from './device-context-provider';
export * from './disclosure';
export * from './dropdown-list';
export * from './dropdown-menu';
export * from './dropdown-menu-button';
export * from './dropdown-navigation';
export * from './error-summary';
export * from './external-link';
export * from './feedbacks';
export * from './field-container';
export * from './fieldset';
export * from './global-banner';
export * from './global-header';
export * from './global-navigation';
export * from './heading';
export * from './icon';
export * from './internationalization-provider';
export * from './label';
export * from './legend';
export * from './link';
export * from './listbox';
export * from './lozenge';
export * from './menu';
export * from './menu-button';
export * from './modal';
export * from './money-input';
export * from './nav-list';
export * from './numeric-input';
export * from './pagination';
export * from './password-creation-input';
export * from './password-input';
export * from './phone-input';
export * from './progress-circle';
export * from './progress-circular';
export * from './progress-indicator';
export * from './progress-tracker';
export * from './radio-button';
export * from './radio-button-group';
export * from './radio-card-group';
export * from './route-link';
export * from './screen-reader-only-text';
export * from './search';
export * from './sectional-banner';
export * from './segmented-control';
export * from './shadow-wrapper';
export * from './side-drawer';
export * from './skip-link';
export * from './slider';
export * from './spinner';
export * from './status';
export * from './stepper-input';
export * from './table';
export * from './tabs';
export * from './tag';
export * from './text-area';
export * from './text-input';
export * from './theme-wrapper';
export * from './toast';
export * from './toggle-switch';
export * from './toggletip';
export * from './tooltip';
export * from './user-profile';
export * from './visually-hidden';
