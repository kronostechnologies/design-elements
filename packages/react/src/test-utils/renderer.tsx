import { ThemeWrapped } from '@design-elements/test-utils/theme-wrapped';
import { mount, ReactWrapper, render } from 'enzyme';
import React, { Component, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DeviceContextProvider } from '../components/device-context-provider/device-context-provider';

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: AllProviders,
    });
}

export function renderWithProviders(
    component: ReactElement,
): Cheerio {
    return render(<AllProviders>{component}</AllProviders>);
}

function AllProviders({ children }: { children: ReactElement }): ReactElement {
    return (
        <MemoryRouter>
            <DeviceContextProvider>
                {ThemeWrapped(children)}
            </DeviceContextProvider>
        </MemoryRouter>
    );
}
