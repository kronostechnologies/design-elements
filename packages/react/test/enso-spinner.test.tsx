import React from 'react';
import renderer from 'react-test-renderer';
import { EnsoSpinner } from '../src/components/enso-spinner';

describe('Enso Spinner', () => {
    test('Matches the snapshot', () => {
        const tree = renderer.create(
            <EnsoSpinner />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
