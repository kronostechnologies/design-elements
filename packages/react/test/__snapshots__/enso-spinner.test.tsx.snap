// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enso Spinner Matches the snapshot 1`] = `
.c0 {
  -webkit-animation: roll 1s infinite;
  animation: roll 1s infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  fill: #e2732d;
  height: 80px;
  width: 83px;
}

<svg
  className="c0"
/>
`;
